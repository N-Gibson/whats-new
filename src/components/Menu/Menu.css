* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.nav-button {
  height: 100%;
  width: 20%;
}