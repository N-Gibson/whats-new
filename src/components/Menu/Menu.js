import React, { Component } from 'react';
import './Menu.css'

// MENU COMPONENT CODE GOES HERE
class Menu extends Component {
  constructor() {
    super()
  }
  
  chooseNews = (event) => {
    event.preventDefault();
    this.props.pickNews(event.target.dataset.category)
  }

  render() {
    return (
      <nav>
        <button 
          data-category='local' 
          onClick={event => this.chooseNews(event)}>Local
        </button>
        <button 
          data-category='technology' 
          onClick={event => this.chooseNews(event)}>Technology
        </button>
        <button 
          data-category='entertainment' 
          onClick={event => this.chooseNews(event)}>Entertainment
        </button>
        <button 
          data-category='science' 
          onClick={event => this.chooseNews(event)}>Science
        </button>
        <button 
          data-category='health' 
          onClick={event => this.chooseNews(event)}>Health
        </button>
      </nav>
    )
  }
}

export default Menu;