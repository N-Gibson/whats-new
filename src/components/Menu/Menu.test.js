import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';

describe('Menu', () => {
  it('should match the snapshot with all of the correct data', () => {
    const wrapper = shallow(<Menu pickNews={jest.fn()}/>)

    expect(wrapper).toMatchSnapshot();
  });
})