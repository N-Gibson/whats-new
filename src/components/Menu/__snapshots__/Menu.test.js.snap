// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot with all of the correct data 1`] = `ShallowWrapper {}`;
